filebeat/prospector/config.go

```diff
@@ -53,3 +55,25 @@ func (config *prospectorConfig) Validate() error {
 
 	return nil
 }
+
+func (config *prospectorConfig) resolvePaths() error {
+	var paths []string
+	if !config.recursiveGlob {
```

Comment:
The reason this is not working is that `recursiveGlob` is a private variable and golang can only extract values in global variables. You must set it to `RecursiveGlob` in prospectorConfig.